js/graphql.md

```diff
@@ -0,0 +1,1346 @@
+---
+---
+# The GraphQL Transform
+
+The GraphQL Transform is a library that simplifies the process of developing, deploying, and maintaining GraphQL APIs. With it, you define your API using the GraphQL Schema Definition Language (SDL) and can then use this library where to transform it into a fully descriptive cloudformation template that implements the API's data model.
+
+The Transform can be run as an independent library, however it is integrated into the Amplify CLI via the `API` category for you to use immediately.
+
+## Quick Start
+
+Navigate into the root of a JavaScript, iOS, or Android project and run:
+
+```bash
+amplify init
+```
+
+Follow the wizard to create a new app. After finishing the wizard run:
+
+```bash
+amplify add api
+
+# Select the graphql option and when asked if you 
+# have a schema, say No.
+# Select one of the default samples. You can change it later.
+# Choose to edit the schema and it will open your schema.graphql in your editor.
+```
+
+You can leave the sample as is or try this schema.
+
+```
+type Blog @model {
+  id: ID!
+  name: String!
+  posts: [Post] @connection(name: "BlogPosts")
+}
+type Post @model {
+  id: ID!
+  title: String!
+  blog: Blog @connection(name: "BlogPosts")
+  comments: [Comment] @connection(name: "PostComments")
+}
+type Comment @model {
+  id: ID!
+  content: String
+  post: Post @connection(name: "PostComments")
+}
+```
+
+Once you are happy with your schema, save the file and click enter in your
+terminal window. If now error messages are thrown then you are good to go
+and can deploy your new API.
+
+```bash
+amplify push
+```
+
+Go to AWS CloudFormation to view it. You can also find your project assets in the amplify/backend folder under your API.
+
+Once the API is finsihed deploying, try going to the AWS AppSync console and
+running some of these queries in your new API's query page.
+
+```
+# Create a blog. Remember the returned id.
+# Provide the returned id as the "blogId" variable.
```

Comment:
Is the syntax by which we add variables in the console _definitely_ clear to someone who has never used AppSync or GraphiQL before? I'm not sure it is. I think we should give some direction, like:

```
# Provide the returned id as the "blogId" variable by 
# entering the following in the "variables" pane of the editor:
{
  "blogId": "returned-id-goes-here"
}
```

Reply:
+1